public/app/features/dashboard/dashgrid/PanelHeader/PanelHeaderTitleItems.tsx

```diff
@@ -0,0 +1,124 @@
+import { css, cx } from '@emotion/css';
+import React from 'react';
+
+import {
+  PanelModel,
+  renderMarkdown,
+  ScopedVars,
+  LinkModelSupplier,
+  PanelData,
+  InterpolateFunction,
+  GrafanaTheme2,
+} from '@grafana/data';
+import { getTemplateSrv } from '@grafana/runtime';
+import { Dropdown, Icon, Menu, ToolbarButton, Tooltip, useStyles2 } from '@grafana/ui';
+
+import { PanelHeaderNotices } from './PanelHeaderNotices';
+
+export interface Props {
+  innerHeight: number;
+  innerWidth: number;
+  panelDescription?: string;
+  alertState?: string;
+  scopedVars?: ScopedVars;
+  replaceVariables: InterpolateFunction;
+  links?: LinkModelSupplier<PanelModel>;
+  data: PanelData;
+  panelId: number;
+}
+
+export function PanelHeaderTitleItems(props: Props) {
+  const { panelDescription, alertState, scopedVars, links, replaceVariables, data, panelId } = props;
+  const styles = useStyles2(getStyles);
+  // description
+  const rawDescription = panelDescription || '';
+  const descriptionMarkdown = getTemplateSrv().replace(rawDescription, scopedVars);
+  const description = renderMarkdown(descriptionMarkdown);
+
+  // panel links
+  const panelLinks = links && links.getLinks(replaceVariables);
```

Comment:
Similar to comment from @torkelo about markdown - the links can be rendered when necessary, i.e. dropdown is opened. A similar thing is implemented in the Data links context menu.